aptos-move/move-examples/token_objects/ambassador/move/sources/ambassador.move

```diff
@@ -0,0 +1,304 @@
+/// This module is used to create ambassador tokens. Ambassador tokens are
+/// soulbound tokens that are minted by the creator of a collection.
+module ambassador_token::ambassador {
+    use std::error;
+    use std::option;
+    use std::string::{Self, String};
+    use std::signer;
+
+    use aptos_framework::object::{Self, Object};
+    use aptos_token_objects::collection;
+    use aptos_token_objects::token;
+    use aptos_token_objects::property_map;
+    use std::bcs;
+    use aptos_framework::event;
+
+    /// The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+    /// Attempted to burn a non-burnable token
+    const ETOKEN_NOT_BURNABLE: u64 = 4;
+    /// Attempted to mutate a property map that is not mutable
+    const EPROPERTIES_NOT_MUTABLE: u64 = 5;
+    // The collection does not exist
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 6;
+
+    /// The ambassador token collection name
+    const COLLECTION_NAME: vector<u8> = b"Ambassador Collection Name";
+    /// The ambassador token collection description
+    const COLLECTION_DESCRIPTION: vector<u8> = b"Ambassador Collection Description";
+    /// The ambassador token collection URI
+    const COLLECTION_URI: vector<u8> = b"Ambassador Collection URI";
+
+    /// The ambassador token collection
+    struct AmbassadorCollection has key {
+        /// Used to mutate collection fields
+        mutator_ref: collection::MutatorRef,
+    }
+
+    /// The ambassador token
+    struct AmbassadorToken has key {
+        /// Used to burn.
+        burn_ref: token::BurnRef,
+        /// Used to control freeze.
+        transfer_ref: object::TransferRef,
```

Comment:
Do we need to keep the token’s transfer ref at all if these tokens are all soul bound

Reply:
Removed.